_posts/2022-05-25-banco-do-brasil-open-banking-user-story.adoc

```diff
@@ -0,0 +1,243 @@
+
+
+=== Multiple persistence units with Hibernate
+
+This feature was very important to our project. The *Open Banking Integrator* has some business data stored in an Oracle database. To improve the speed of processing, there is some information that we need to obtain directly from this database. As our application has a DB2 database, the Hibernate Multiple Persistence units feature was very helpful. More information is available on the https://quarkus.io/guides/hibernate-orm#multiple-persistence-units[Hibernate Quarkus Guide].
+
+
+=== Fault tolerance Retry 
+
+The APIs we execute can return errors saying that the system is temporary unavailable. In this situation, we need to wait some time and try again.
+
+This is very simple when using the Retry:
+
+[source, java]
+----
+@Retry(retryOn = { ExceptionOfTheApiThatWeMustTryAgain.class }, maxRetries = 3, delay = 1000)
+public void callExternalEndpoint() {
+  // 
+}
+----
+
+We can configure the maxRetries and the delay between the retries on our properties:
+[source, properties]
+----
+Retry/maxRetries=${APP_MAX_RETRY:3}
+Retry/delay=${APP_DELAY_MS_RETRY:1000}
+----
+
+The https://download.eclipse.org/microprofile/microprofile-fault-tolerance-2.1/microprofile-fault-tolerance-spec.html#configuration[MicroProfile Fault Tolerance specification guide] explains that we can configure these values individually (class or method) or globally.
+
+This configuration is very useful because we don't want the retries to take too much time in our tests, so it is possible to set values specifically for the tests:
+
+[source, properties]
+----
+%test.Retry/maxRetries=${APP_MAX_RETRY:3}
+%test.Retry/delay=${APP_DELAY_MS_RETRY:1}
+----
+
+
+=== Synchronizing access with fault tolerance Bulkhead
+
+Our *scheduler* microservice can't run the schedule in parallel. 
+
+We deploy it on Kubernetes with the strategy type *Recreate*, meaning that all pods are terminated before a new one is created. This guarantees that only one pod at a time will be executed. So how do we avoid multiple threads from the same instance executing the same method at the same time?
+
+The answer is to use the Bulkhead fault tolerance.
+
+[source, java]
+----
+// maximum 1 concurrent requests allowed, maximum 1 requests allowed in the waiting queue
+@Bulkhead(value = 1, waitingTaskQueue = 1)
+public void processSchedule() {
+  // 
+}
+----
+
+When a request cannot be added to the waiting queue, BulkheadException will be thrown.
+
+
+
+== Final words about Quarkus
+
+Our organization has hundreds of Java programmers, but one problem that we see is the difficulty to hire people who already have experience with Quarkus. Although, we believe that a person who has experience with other Java frameworks can learn Quarkus very fast and, with time, we will have more professionals mastering it.
```

Comment:
```suggestion
Our organization has hundreds of Java programmers, but one problem that we see is the difficulty to hire people who already have experience with Quarkus. That being said, we believe that a person who has experience with other Java frameworks can learn Quarkus very fast and, with time, we will have more professionals mastering it.
```

Reply:
@gsmet , @maxandersen [also suggested to change this sentence](https://github.com/quarkusio/quarkusio.github.io/pull/1444#discussion_r882451294)

After accepting his suggestion, now it's like this:
> Our organization has hundreds of Java programmers, but one problem that we see is the difficulty to hire people who already have experience with Quarkus. **We though** believe that a person who has experience with other Java frameworks can learn Quarkus very fast and, with time, we will have more professionals mastering it.

Should I change to your suggestion? Both look good to me.